### ISSUE TITLE
Replace XGBoost with GradientBoostingClassifier

### ISSUE THREAD

**Issue by rhiever** [rhiever/tpot#151]
Finally finished the sklearn benchmarks, and one of the results is that XGBoost doesn't really seem to perform significantly better than sklearn's GradientBoostingClassifier on the benchmarks we're using.

Considering that XGBoost has caused many people to have TPOT install issues, let's replace XGBoost with sklearn's GradientBoostingClassifier and drop XGBoost as a dependency until they get their install issues sorted out.

**Comment by rhiever** [comment]
@teaearlgraycold, you down for this issue?

**Comment by teaearlgraycold** [comment]
Sure

**Comment by rhiever** [comment]
Thanks. Maybe we can piggyback it in #147.

**Comment by KhaoticMind** [comment]
Can't we keep xgboost in the pipeline, but not explicity it as a dependency?

So at runtime, it is used if it's present, otherwise we fall back to sklearn implementation.
In my experience xgboost uses A LOT less memmory than GradientBoostingClassifier. I might try to come up with some benchmarks from kaggle competitions to prove the point of keeping XGBoost in.

**Comment by rhiever** [comment]
Happy to see said benchmarks, @KhaoticMind.

It's possible that we could rework TPOT to make it an optional dependency and do as you said.

**Comment by KhaoticMind** [comment]
Hi @rhiever , while I couldn't do a Memory wise benchmark, I did do have some CPU based results where you can easily see XGB performs A LOT better than GradientBoostingClassifier (about 4x times better with the same conditions).
https://www.kaggle.com/khaoticmind/expedia-hotel-recommendations/analysis-of-time/run/245245/log

Check the code here: 
https://www.kaggle.com/khaoticmind/expedia-hotel-recommendations/analysis-of-time/run/245245

Note that I've set nthreads=1 in XGBoost, to make it comparable with sklearn inplementation (that does not support multithread).
If you see any errors on the code, please point than out and I can re-run the tests.

**Comment by rhiever** [comment]
That's interesting that XGBoost is so much faster. Do you know if it performs better, though? On the 155 data set benchmark that I have, it XGBoost didn't really perform better accuracy-wise.

We're going to replace XGBoost with the sklearn GradientBoostingClassifier for now, and raise a separate issue (#154) to make XGBoost an optional dependency in the future. @KhaoticMind, I welcome you to contribute that code, else we'll have it on our TODO stack for a while.
